android/src/main/kotlin/com/example/posthog_flutter/PosthogFlutterPlugin.kt

```diff
@@ -0,0 +1,310 @@
+            }
+
+            "enable" -> {
+                enable(result)
+            }
+
+            "isFeatureEnabled" -> {
+                isFeatureEnabled(call, result)
+            }
+
+            "reloadFeatureFlags" -> {
+                reloadFeatureFlags(result)
+            }
+
+            "group" -> {
+                group(call, result)
+            }
+
+            "getFeatureFlag" -> {
+                getFeatureFlag(call, result)
+            }
+
+            "getFeatureFlagPayload" -> {
+                getFeatureFlagPayload(call, result)
+            }
+
+            "getFeatureFlagAndPayload" -> {
+                getFeatureFlagAndPayload(call, result)
+            }
+
+            "register" -> {
+                register(call, result)
+            }
+
+            else -> {
+                result.notImplemented()
+            }
+        }
+
+    }
+
+    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
+    }
+
+    private fun getFeatureFlag(call: MethodCall, result: Result) {
+        try {
+            val featureFlagKey: String? = call.argument("key")
+            val value: Any? = PostHog.getFeatureFlag(featureFlagKey!!)
+            result.success(value)
+        } catch (e: Exception) {
+            result.error("PosthogFlutterException", e.localizedMessage, null)
+        }
+    }
+
+    private fun getFeatureFlagPayload(call: MethodCall, result: Result) {
+        try {
+            val featureFlagKey: String? = call.argument("key")
+            val value: Any? = PostHog.getFeatureFlagPayload(featureFlagKey!!)
+            result.success(value)
+        } catch (e: Exception) {
+            result.error("PosthogFlutterException", e.localizedMessage, null)
+        }
+    }
+
+    private fun getFeatureFlagAndPayload(call: MethodCall, result: Result) {
+        try {
+            val featureFlagKey: String? = call.argument("key")
+            val status: Any? = PostHog.getFeatureFlag(featureFlagKey!!)
+            val payload: Any? = PostHog.getFeatureFlagPayload(featureFlagKey)
+            val featureAndPayload = mutableMapOf<String, Any?>()
+
+            when (status) {
+                null -> {
+                    featureAndPayload["isEnabled"] = false
+                }
+
+                is String -> {
+                    featureAndPayload["isEnabled"] = true
+                    featureAndPayload["variant"] = status
+                }
+
+                else -> {
+                    featureAndPayload["isEnabled"] = status
+                }
+            }
+            featureAndPayload["data"] = payload
+
+            result.success(featureAndPayload)
+        } catch (e: Exception) {
+            result.error("PosthogFlutterException", e.localizedMessage, null)
+        }
+    }
+
+    private fun identify(call: MethodCall, result: Result) {
+        try {
+            val userId: String? = call.argument("userId")
```

Comment:
```suggestion
            val userId = call.argument("userId") as? String
```

I'd prefer to use Kotlin's type inference.